src/lib.rs

```diff
@@ -432,14 +432,9 @@ where
         .construct()
 }
 
-// test that ReadHandle isn't Sync
-// waiting on https://github.com/rust-lang/rust/issues/17606
-//#[test]
-//fn is_not_sync() {
-//    use std::sync;
-//    use std::thread;
-//    let (r, mut w) = new();
-//    w.insert(true, false);
-//    let x = sync::Arc::new(r);
-//    thread::spawn(move || { drop(x); });
-//}
+
+#[test]
+fn is_not_sync() {
+    extern crate static_assertions as sa;
+    sa::assert_not_impl_all!(ReadHandle<(), ()>: Send, Sync);
```

Comment:
Here's the docs for the macro

https://docs.rs/static_assertions/1.1.0/static_assertions/macro.assert_not_impl_all.html

Reply:
Hmm, this doesn't seem quite right. `ReadHandle` _should_, I believe, be `Send`, it just should _not_ be `Sync`.

Reply:
Yeah so I found the documentation a little confusing. But I believe this is correct given the following:

```
Although u32 implements From<u16>, it does not implement Into<usize>:

assert_not_impl_all!(u32: From<u16>, Into<usize>);
```

I will take a look at breaking the test and get back to you tomorrow.

per: https://docs.rs/static_assertions/1.1.0/static_assertions/macro.assert_not_impl_all.html

Reply:
Reading that, I _think_ the meaning is to check that it does not implement _all_ of the passed traits. So in the above, the test is "`u32` does not implement _both_ `From<u16>` _and_ `Into<usize>`". I think I would just write two assertions. One that the type _is_ `Send`, and on that the type is _not_ `Sync`.